### ISSUE TITLE
debugging calls to `modifyStream` that take too long

### ISSUE THREAD

**Issue by little-dude** [pstavirs/ostinato#178]
Hi,

I'm writing a library that hides all the protocol buffer stuff, and that has auto-generated documentation: https://github.com/little-dude/simple-ostinato/

It's still work in progress and not yet available on pypi, but I'm running into a weird issue, that I don't know how to debug. On travis, the tests take about ~5 minutes to run whereas they take less than 10 seconds locally (tried on Alpine Linux and Archlinux). See this run for instance: https://travis-ci.org/little-dude/simple-ostinato/builds/125923860

I appears that some calls to `modifyStream` take between 45 and 50 seconds. Here are the logs:

```
modifying stream. Time elapsed: 0
[127.0.0.1:57849] Server(on_clientSock_dataAvail): successfully received/parsed msg <----
[127.0.0.1:57849] method = 7
[127.0.0.1:57849] req = 
[127.0.0.1:57849] port_id {
[127.0.0.1:57849]   id: 3
[127.0.0.1:57849] }
[127.0.0.1:57849] stream {
[127.0.0.1:57849]   stream_id {
[127.0.0.1:57849]     id: 0
[127.0.0.1:57849]   }
[127.0.0.1:57849]   core {
[127.0.0.1:57849]   }
[127.0.0.1:57849]   control {
[127.0.0.1:57849]   }
[127.0.0.1:57849]   protocol {
[127.0.0.1:57849]     protocol_id {
[127.0.0.1:57849]       id: 100
[127.0.0.1:57849]     }
[127.0.0.1:57849]     [OstProto.mac] {
[127.0.0.1:57849]       dst_mac: 0
[127.0.0.1:57849]       dst_mac_mode: e_mm_fixed
[127.0.0.1:57849]       dst_mac_count: 16
[127.0.0.1:57849]       dst_mac_step: 1
[127.0.0.1:57849]       src_mac: 0
[127.0.0.1:57849]       src_mac_mode: e_mm_fixed
[127.0.0.1:57849]       src_mac_count: 16
[127.0.0.1:57849]       src_mac_step: 1
[127.0.0.1:57849]     }
[127.0.0.1:57849]   }
[127.0.0.1:57849] }
[127.0.0.1:57849] ---->
[127.0.0.1:57849] In virtual void MyService::modifyStream(google::protobuf::RpcController*, const OstProto::StreamConfigList*, OstProto::Ack*, google::protobuf::Closure*)
[127.0.0.1:57849] In updatePacketListInterleaved
[127.0.0.1:57849] minGap   = 18446744073709551615
[127.0.0.1:57849] duration = 1000000000

# this is where drone seems to stay stuck

[127.0.0.1:57849] loop Delay = 1/0
[127.0.0.1:57849] Server(sendRpcReply): sending 8 bytes to client <----
[127.0.0.1:57849] 0002000700000000
[127.0.0.1:57849] method = 7
[127.0.0.1:57849] req = 
[127.0.0.1:57849] ---->
done modifying stream. Time elapsed: -44
```

Do you have any pointer for further debugging? Unfortunately, since I cannot reproduce locally I cannot debug with pdb, or use a custom build for drone...

**Comment by little-dude** [comment]
Oops I forgot to mention that the OS is Ubuntu 14.04.3 LTS, and ostinato is 0.7.1-2, installed as follow:
```
sudo sh -c "echo 'deb http://download.opensuse.org/repositories/home:/pstavirs:/ostinato/xUbuntu_14.04/ /' >> /etc/apt/sources.list.d/ostinato.list"
wget http://download.opensuse.org/repositories/home:pstavirs:ostinato/xUbuntu_14.04/Release.key ; sudo apt-key add - < Release.key
sudo apt-get -qq update
sudo apt-get install -y --force-yes ostinato
```

**Comment by pstavirs** [comment]
@little-dude: simple-ostinato will surely make it much more accessible for new comers and maybe even more pythonic than python-ostinato.

I didn't get a chance to investigate this, but I did notice that you seem to be using interleaved mode. Could you try sequential streams instead? https://github.com/pstavirs/ostinato/wiki/UserGuide#transmit-mode

**Comment by little-dude** [comment]
Ok I finally found out what the issue was. You're right, the port was configured iwth interleaved mode. This should not have been the case, because normally, drone is restarted between tests, so the configuration should be gone.

After debugging I found out that my test teardown in which I kill drone was not run on travis. Fixing this fixed the problem. Sorry for the noise...

I'll send an email to the ostinato mailing list if/when `simple-ostinato` becomes usable for real world scenarios :)
